### DISCUSSION TITLE
linux系统建议使用fcitx5-rime（而非fcitx-rime）

### DISCUSSION THREAD

**Discussion by gaboolic** [rime/home#1493] [General]
ibus-rime最新的包是on Jan 24, 2021
fcitx-rime最新的包是2017年的
都是多年不维护了
fcitx5-rime 最新的[5.1.5](https://github.com/fcitx/fcitx5-rime/releases/tag/5.1.5)  两周前打的tag
官网上的说明太老，我踩坑了，建议更新一下 让大家用fcitx5-rime

**Comment by gaboolic** [comment]
https://fcitx-im.org/wiki/Install_Fcitx_5/zh-cn

**Comment by yanhuacuo** [comment]
你说的完全不符合实际，事实上，即使是最保守的 debian12，也有较新的，完整支持 lua 组件的 ibus-rime 及 librime-plugin-lua 插件。

ibus-rime 本身没有什么变化，变化的是 librime 部分：https://packages.debian.org/bookworm/librime1

比如：librime-plugin-lua（1.8.5+dfsg1~git20230115-1），实际更新于去年11月份，librime1.8.5 也是较新的版本。


另一层面，ibus-rime 要比 fcitx5-rime 的兼容性更好，比如 Qt 类程序中，fcitx5 就要特别编译额外的输入法动态链接库。


ibus-rime 还有个神仙项目：https://github.com/hchunhui/ibus-rime.AppImage

一键运行，非常易用。

**Comment by gaboolic** [comment]
我用的是linuxmint系统，基于ubuntu 22.04.4 LTS
默认是用的fcitx5，安装fcitx5-rime版本是5.0.11-1 https://pkgs.org/download/fcitx5-rime
我听了你的试了下ibus，装了ibus-rime 版本是1.5.0-1 https://pkgs.org/download/ibus-rime
这两个版本对我来说都有问题，lua似乎是能用一部分，版本比较低，然后字频词频完全不对。ibus-rime在我这还有个额外的问题是在github打字看不到候选框。（虽然我用双拼+形不太需要候选框，但是打 基于 这个词的时候居然在第二页，词频不对）我看了下这两个版本最新更新都是2022年
至于‘ibus-rime 还有个神仙项目：https://github.com/hchunhui/ibus-rime.AppImage’  似乎是个人维护的，不是官方的也不是ibus打的，可能不稳定，也可能随时撂挑子。

所以我认为非arch系最好的方式还是使用flatpak的fcitx5-rime，是最新的

**Reply by yanhuacuo** [reply]
词频不对，是你的配置不对，跟ibus-rime没有关系。lua版本并不低，以及，librime-lua 这个件的作者做的 [/ibus-rime.AppImage](https://github.com/hchunhui/ibus-rime.AppImage%E2%80%99)  这个项目。

在 Gnome 下， ibus-rime 没有任何兼容性问题。

**Reply by gaboolic** [reply]
你不要把别人当傻子看待啊，什么叫我的配置不对。我都说了是版本问题，我的发行版默认的源无论ibus-rime还是fcitx5-rime版本都比较低大概是2022年的包。我使用flatpak最新的fcitx5-rime版本就ok了。而且我又不是只有linux设备，mac和windows也都是ok的。
我的发行版ibus-rime和fcitx5-rime版本低 自然lua版本就低，有的函数能用有的不能用，比如输入time打出日期是好的，但是一些方案里有复杂的lua就不能用。
“在 Gnome 下， ibus-rime 没有任何兼容性问题。”至于这一点，我想说的是桌面有很多种。很多发行版都是默认fcitx5的。通过ibus和fcitx5的各种评价来看，fcitx5的评价更高

**Reply by gaboolic** [reply]
![ibus](https://github.com/rime/home/assets/3831173/79403f17-3057-4137-bb16-0cf860035e37)
ibus-rime的效果。字频不对，候选框位置不对

**Reply by gaboolic** [reply]
![fcitx5](https://github.com/rime/home/assets/3831173/59dde534-e272-4ba1-b868-aa5edfaba313)
使用flatpak安装的最新版fcitx5-rime，完全正常。默认源里的fcitx5-rime字频不对，候选框是正常的，能用。

**Reply by gaboolic** [reply]
旧版本的对带辅助码的dict文件词频是有问题的https://github.com/rime/home/issues/1046

**Comment by gaboolic** [comment]
flathub上的fcitx5版本到最新了
https://flathub.org/zh-Hans/apps/org.fcitx.Fcitx5

**Comment by lotem** [comment]
分享自己的經驗。很好。

Linux 系統生態複雜，發行版衆多，軟件包的支持情況也不盡相同。
老牌的軟件，如果不出現和其他系統組件不兼容的問題，那麼他在各 Linux 發行版之間的普及是最好的。
相反，新出的軟件可能在其他 Linux 發行版上還沒有得到很好的支持，或者說無法在默認的配置下方便地使用。

歸功於 librime 與 ibus-rime 對輸入法前後端的分工，大部分功能升級只需通過更新 librime 完成。ibus-rime 可以長期保持穩定兼容。

很多 Linux 發行版上默認配置了 IBus 輸入框架。綜合來看，ibus-rime 現在還是任意 Linux 系統上最易上手的選項。

**Reply by gaboolic** [reply]
ubuntu系的ibus-rime和fcitx5-rime倒是都有，但是都比较老，有词频问题和lua脚本问题。没办法我只能装了flatpak版本的fcitx5-rime，这样能用到最新的librime

**Reply by gaboolic** [reply]
不过如果等ubuntu24.04出了，估计会更新ibus-rime和fcitx5-rime吧……

**Reply by gaboolic** [reply]
ibus-rime是因为tags 1.5.0 on Jan 24, 2021，代码最后更新也停在3 years ago，然后我apt install ibus-rime也是1.5.0 但是词频有问题，让我产生了ibus-rime不维护的感觉。

**Reply by yanhuacuo** [reply]
事实上，只有 librime-lua 这个插件的开发者，才会第一时间把 librime-lua 最新的版本放到这个项目中来: 

https://github.com/hchunhui/ibus-rime.AppImage

你用到的 lua 插件，就是这位作者写的：https://github.com/hchunhui/librime-lua

**Reply by lotem** [reply]
Ubuntu xxx.x 的體驗不能代表 Linux。
另外你講的那些問題都跟 ibus-rime 無關，是 librime 和數據文件決定的。無論前端用 ibus-rime 還是 fcitx5-rime 都沒有區別。

樓主的論點可以歸結爲：
安裝 flatpak 包可以用到目前打包了最新版本的 Rime 軟件，解決了樓主關注的兩個問題。
但要論是否應該建議 Linux 用戶都用這種方法，還要回答幾個問題：
用戶的 Linux 發行版是否能方便地安裝 flatpak；
用戶是否認爲系統包管理器提供的較舊版本的 Rime 軟件包，存在的問題影響輸入體驗，值得用一種非默認的安裝方式升級；
目前的狀態是 flatpak 的版本更新；然而維護這個包的團隊是否有足夠的人力長期保持相對於各大 Linux 發行版軟件包的領先發佈？

**Reply by gaboolic** [reply]
没错，我会更新一下我的观点。
”另外你講的那些問題都跟 ibus-rime 無關，是 librime 和數據文件決定的。無論前端用 ibus-rime 還是 fcitx5-rime 都沒有區別。“ 不过ibus-rime的前端在我这里有一个额外的问题是候选框乱飘，后面我装一下[ibus-rime.AppImage] 看看能否解决

**Reply by gaboolic** [reply]
flatpak 的版本更新，如果相对于ubuntu發行版軟件包的領先發佈，我认为会的。
相对于Arch系 不一定，可能差不多。所以ubuntu系 可以用flatpak版本

**Comment by gaboolic** [comment]
更新一下我的观点。

fcitx5-rime：
fcitx-rime最新的包是2017年的，多年不维护了，建议使用fcitx5-rime，最新的[5.1.5](https://github.com/fcitx/fcitx5-rime/releases/tag/5.1.5) 两周前打的tag。
如何能使用到最新的fcitx5-rime，可以看https://github.com/fcitx/fcitx5  里面有各大发行版中fcitx5的版本，如果版本太低不满意，可以使用flatpak版本的fcitx5 https://fcitx-im.org/wiki/Install_Fcitx_5/zh-cn

ibus-rime：
github上最新的tag是1.5.0 Jan 24, 2021，但并不代表不维护了，大部分功能升級只需通過更新 librime 完成。ibus-rime 可以長期保持穩定兼容。
例如ibus-rime_1.5.0-3  是2023-02-08上传的，（Rebuild against librime 1.8.5. ）会升级librime版本
如果发行版中的ibus-rime版本低，可以使用AppImage版的https://github.com/hchunhui/ibus-rime.AppImage

**Comment by gaboolic** [comment]
我的标题主要针对的是fcitx-rime
因为之前官网上没写fcitx5-rime 只写了fcitx-rime， 我不知道有fcitx5-rime
